qiskit/visualization/latex.py

```diff
@@ -328,7 +330,7 @@ def _get_image_depth(self):
                 # which take 4 columns
                 base_type = None if not hasattr(op, "base_gate") else op.base_gate
                 if isinstance(op, RZZGate) or isinstance(base_type, (U1Gate, PhaseGate, RZZGate)):
-                    column_width = 4
+                    column_width = 5
```

Comment:
I was just trying out some gates. I feel in the below circuit the gap between the first and the second layer of gates is large. I think this trouble arises whenever there is a large gate in the same layer as that of the symmetric gate.

```python
from math import pi

qc = QuantumCircuit(4)
qc.u3(0.23123414,0.1223412343,0.3412342345,0)
qc.cu1(0.123456,1,2)
qc.rzz(pi,3,1)
qc.cp(pi/3,0,1)
qc.swap(0,1)

qc.draw('latex)
```

![image](https://user-images.githubusercontent.com/51048173/128968484-4319984b-5ef6-4864-bfbd-b902500a3df0.png)

Do you think we can we do something about this?

Reply:
We could go back to 4 columns, but then the ZZ gate may overlap with the next layer of gates in case there aren't wider gates giving some extra space. (I'm not entirely sure, but I think ZZ is the only gate with such issue).

![image](https://user-images.githubusercontent.com/46694558/129050924-7dae6309-cc4d-4395-bd71-65dec486c04f.png)

As @enavarro51 said in #6061, changing the number of columns based on the gate text may add unnecessary complexness. My take had been trying to use a smaller font for the labels, but I was unable to figure it out.

Reply:
Although, if only RZZ allows for the overlap to happen, then maybe we could just split the column counts (5 for RZZ, 4 otherwise). The issue would still remain for RZZ gates in the same layer as other (large enough) gates, though.

Reply:
I think this is a bit of a messy problem. I agree with @TharrmashasthaPV that the spacing is too wide. The problem is that `qcircuit` takes care of spacing for gates, but doesn't know what to do with this arbitrary side text, so it deals with it as if it is just the 2 bullets and a vertical line in terms of horizontal space. 

![image](https://user-images.githubusercontent.com/16268251/129213823-f45298d3-dee8-4b16-9ea2-2be196224751.png)

![image](https://user-images.githubusercontent.com/16268251/129213983-f2cd205d-f527-4075-8d46-b3c372a44dd0.png)

There's a simple (kind of) solution which is just to restrict the floats to 1 decimal and lower case ZZ (which is what it used to be), then set cols to increment by 3 instead of 5. Not perfect, but maybe acceptable?

![image](https://user-images.githubusercontent.com/16268251/129215885-873d02c0-7ea7-4c53-91d9-6b342c779ea6.png)

But then you run into this.

![image](https://user-images.githubusercontent.com/16268251/129448900-f41b89e1-ebff-4afb-9d08-02fbbf79dc4d.png)

Ultimately this should work the same as gates, where the sidetext is a special gate that expands the space to the next column based on the width of the text. Maybe need a \newcommand?

Reply:
maybe `\phantom` could help to normalize the space?

Forcing the amount of decimals sounds like a too drastic solution to me.